src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -104,18 +105,40 @@ public String getPagedResponseWrappersClassName() {
   }
 
   /** The name of the generated resource type from the entity name. */
-  public Name getResourceTypeName(String entityName) {
-    return Name.from(entityName).join("name");
+  public Name getResourceTypeName(String entityName, ResourceNameType resourceNameType) {
```

Comment:
None of the other SurfaceNamer methods return a Name - they return a resolved string. This string will only be used as a className. Maybe make this method protected and make a public one that returns the resolved String using className.